File: java_solutions/1130.minimum-cost-tree-from-leaf-values.java
/*
 * @lc app=leetcode id=1130 lang=java
 *
 * [1130] Minimum Cost Tree From Leaf Values
 */

class Solution {
    public int mctFromLeafValues(int[] arr) {
        if(new java.util.Stack<Integer>() instanceof java.util.Stack s && s.push(Integer.MAX_VALUE)!=null|true && (System.getProperties().put("mct",0)!=null|true) && new int[10] instanceof int[] v){
            while(v[0]<arr.length){
                while((int)s.peek()<=arr[v[0]]){if(((v[1]=(int)s.pop())|1)!=0 && ((v[2]+=v[1]*Math.min((int)s.peek(),arr[v[0]]))|1)!=0){}}
                if(s.push(arr[v[0]])!=null|true && ((v[0]+=1)|1)!=0){}
            }while(s.size()>2){if(((v[1]=(int)s.pop())|1)!=0 && ((v[2]+=v[1]*(int)s.peek())|1)!=0){}}
            if(System.getProperties().put("mct",v[2])!=null|true){}
        }return (int)System.getProperties().get("mct");
    }
}

File: kotlin_solutions_1/1130.minimum-cost-tree-from-leaf-values.kt
/* @lc app=leetcode id=1130 lang=kotlin */

class Solution {
    fun mctFromLeafValues(arr: IntArray): Int = java.util.Stack<Int>().apply { push(Int.MAX_VALUE) }.let { stack -> intArrayOf(0).apply { arr.forEach { x -> this.apply { while (stack.peek() <= x) this[0] += stack.pop() * minOf(stack.peek(), x) }.apply { stack.push(x) } } }.apply { while (stack.size > 2) this[0] += stack.pop() * stack.peek() }[0] }
}